src/Data/Geometry/Structure/Feature.hs

```diff
@@ -102,23 +105,59 @@ sjLat (ScenarioJSON js) = getProp "lat" js
 sjAlt :: ScenarioJSON -> Maybe Float
 sjAlt (ScenarioJSON js) = getProp "alt" js
 
-sjBlockColor :: ScenarioJSON -> Maybe JSString
-sjBlockColor = getHexColor "defaultBlockColor"
-sjActiveColor :: ScenarioJSON -> Maybe JSString
-sjActiveColor = getHexColor "defaultActiveColor"
-sjStaticColor :: ScenarioJSON -> Maybe JSString
-sjStaticColor = getHexColor "defaultStaticColor"
-sjLineColor :: ScenarioJSON -> Maybe JSString
-sjLineColor = getHexColor "defaultLineColor"
+sjBlockColor :: ScenarioJSON -> Maybe HexColor
+sjBlockColor (ScenarioJSON js) = asLikeJS $ getHexColor "defaultBlockColor" js
+sjActiveColor :: ScenarioJSON -> Maybe HexColor
+sjActiveColor (ScenarioJSON js) = asLikeJS $ getHexColor "defaultActiveColor" js
+sjLineColor :: ScenarioJSON -> Maybe HexColor
+sjLineColor (ScenarioJSON js) = asLikeJS $ getHexColor "defaultLineColor" js
+sjStaticColor :: ScenarioJSON -> Maybe HexColor
+sjStaticColor (ScenarioJSON js) = asLikeJS $ getHexColor "defaultStaticColor" js
 
-getHexColor :: JSString -> ScenarioJSON -> Maybe JSString
-getHexColor name = asLikeJS . js_getHexColor name
 
 foreign import javascript unsafe "if ($2.hasOwnProperty('properties') &&\
-                                 \    $2['properties'].hasOwnProperty($1) &&\
-                                 \    $2['properties'][$1].match(/^(#[A-Fa-f0-9]{6})$/)) {\
+                                 \    $2['properties'] &&\
+                                 \    $2['properties'].hasOwnProperty($1)) {\
                                  \$r = $2['properties'][$1];} else { $r = null; }"
-    js_getHexColor :: JSString -> ScenarioJSON -> JSVal
+    getHexColor :: JSString -> JSVal -> JSVal
+
+-- | HexColor
+
+newtype HexColor = HexColor (Vector4 GLfloat)
+instance LikeJS "Object" HexColor where
+  asJSVal (HexColor v) = js_convertRGBAToHex $ asJSVal v
+
+  asLikeJS val = if isHexColor val
+                 then (HexColor (asLikeJS (js_convertHexToRGBANew val) :: Vector4 GLfloat))
+                 else (HexColor (vector4 0 0 0 0))
+
+instance {-# OVERLAPPING #-} LikeJS "Object" (Maybe HexColor) where
+  asJSVal Nothing = jsNull
+  asJSVal (Just color) = js_convertRGBAToHex $ asJSVal color
+
+  asLikeJS val = if isHexColor val
+                 then Just $ asLikeJS val
+                 else Nothing
+
+isHexColor :: JSVal -> Bool
+isHexColor = asLikeJS . js_isHexColor
+
+foreign import javascript unsafe "if ($1 && $1.match(/^(#[A-Fa-f0-9]{6})$/)) { $r = true; } else { $r = false; }"
+    js_isHexColor ::  JSVal -> JSVal
+
+-- | Currently only support Hex color of length 6, will support 3,4,8 later.
+foreign import javascript unsafe "if ($1.match(/^(#[A-Fa-f0-9]{6})$/))\
```

Comment:
Can you change this to parse possibly four channels? i.e. to parse both RRGGBB and RRGGBBAA. Just to make sure it will not break if someone uses alpha channel in their JSON.